### PR TITLE
CASSANDRA-16522: Fix flaky ClientRequestSizeMetricsTest

### PR DESCRIPTION
**Pull request by jacek-lewandowski**

### CODE REVIEW

File: test/unit/org/apache/cassandra/Util.java [hunk 1]
```diff
@@ -819,4 +820,38 @@ public static void setUpgradeFromVersion(String version)
...
         Gossiper.instance.expireUpgradeFromVersion();
     }
+
+    public interface ThrowingRunnable
+    {
+        void run() throws Throwable;
+    }
+
+    @SuppressWarnings("ConstantConditions")
+    public static void retry(ThrowingRunnable runnable, Predicate<Throwable> canRetry, int maxRetries, long delay) throws Throwable
```
Comment (michaelsembwever) [inline test/unit/org/apache/cassandra/Util.java]: very much like [Awaitility](http://www.awaitility.org/). (which is used in the project already)

if Awaitility is not applicable, can we get a more informative method name please.

**Review by blerer — COMMENTED** [review]
Rather than using some kind of retry mechanism, the patch could simply disable schema refresh using `session.setSchemaMetadataEnabled(false);`  and renable it afterward.

**Comment by jacek-lewandowski** [comment]
@blerer which driver version has that method?

File: test/unit/org/apache/cassandra/Util.java (hunk 1 above)
Comment (jacek-lewandowski) [reply test/unit/org/apache/cassandra/Util.java]: It is perfectly applicable...

**Comment by blerer** [comment]
Sorry, I was looking at the wrong doc. In our case we should use  `session.getCluster().getConfiguration().getQueryOptions().setMetadataEnabled(false)`

**Review by michaelsembwever — APPROVED** [review]

File: test/unit/org/apache/cassandra/metrics/ClientRequestSizeMetricsTest.java [hunk 2]
```diff
@@ -65,24 +65,34 @@ public static void setUp()
...
-        assertThat(responseLength).isGreaterThan(0);
-
-        checkMetrics(1, requestLength, responseLength);
-
-        // Let's fire the same request again and test that the changes are the same that previously
-        executeNet(version, "SELECT * from system.peers");
-
-        checkMetrics(2, requestLength, responseLength);
+        // It may happen that the schema refreshment is done in the middle of the test which can pollute the results
+        // We will retry if we collect more then expected messages hoping that it was due that
```
Comment (adelapena) [inline test/unit/org/apache/cassandra/metrics/ClientRequestSizeMetricsTest.java]: Does this comment about retrying come from a previous approach?

Comment (JeremiahDJordan) [inline test/unit/org/apache/cassandra/metrics/ClientRequestSizeMetricsTest.java]: comment was not updated when switching to using setMetadataEnabled

**Review by adelapena — APPROVED** [review]
Looks good to me. [Here](https://jenkins-dse.build.dsinternal.org/view/Parameterized/job/parameterized-testall/708/) are 100 successful runs in the multiplexer.

File: test/unit/org/apache/cassandra/metrics/ClientRequestSizeMetricsTest.java (hunk 2 above)
Comment (michaelsembwever) [reply test/unit/org/apache/cassandra/metrics/ClientRequestSizeMetricsTest.java]: It was. I'm going to remove it and commit.

**Pull request closed without merging**